src/domain/core/src/entities/metadata_chain.rs

```diff
@@ -106,6 +106,34 @@ pub trait MetadataChainExt: MetadataChain {
         }
     }
 
+    /// Same as [Self::last_of_type_ref] defaulting to the head reference
+    async fn last_of_type<E: VariantOf<MetadataEvent>>(
```

Comment:
Just a note: our UI would need to filter pages of N elements of type to implement filters properly. See https://github.com/kamu-data/kamu-web-ui/issues/122

Reply:
Yeah, I think we need and API like `iter_blocks` that accepts a mask of event types, e.g.

```rust
metadata_chain.iter_blocks_ref(
  &BlockRef::Head, 
  MetadataEventType::SetPollingSource | MetadataEventType::SetLicense | MetadataEventType::SetInfo
)`
```
This would let `MetadataChain` implementations decide which speedups they can use to make traversal most efficient. The `last_of_type` can then be a helper that delegates to such function.

I was going to add such function first, but my PR was already growing out of control.